src/layers/legacy/medVtkInria/vtkImageView/vtkImageView2D.cxx

```diff
@@ -1743,16 +1745,19 @@ void vtkImageView2D::SetInput(vtkAlgorithmOutput* pi_poVtkAlgoOutput, vtkMatrix4
 void vtkImageView2D::SetInputLayer(vtkAlgorithmOutput* pi_poVtkAlgoOutput, vtkMatrix4x4 *matrix /*= 0*/, int layer /*= 0*/)
 {
     // layer > 0
-
     this->AddLayer(layer);
     pi_poVtkAlgoOutput = this->ResliceImageToInput(pi_poVtkAlgoOutput, matrix);
-
     vtkImage2DDisplay *imageDisplay = this->GetImage2DDisplayForLayer(layer);
-    imageDisplay->SetInputProducer(pi_poVtkAlgoOutput);
-    imageDisplay->SetInputData(static_cast<vtkImageAlgorithm*>(pi_poVtkAlgoOutput->GetProducer())->GetOutput());
-    imageDisplay->GetImageActor()->SetUserMatrix (this->OrientationMatrix);
-
-    this->SetColorRange(imageDisplay->GetMedVtkImageInfo()->scalarRange, layer);
+    if (imageDisplay)
```

Comment:
is imageDisplay could be null? If yes, it should be a problem higher perhaps?

Reply:
I did not understood. image is has be retrieved by GetImage2DDisplayForLayer which could return a null ptr.

Reply:
I mean, GetImage2DDisplayForLayer is set in these files, and if it's null here, we have an algorithmic problem i think, and could be solved. The test on GetImage2DDisplayForLayer or GetImage3DDisplayForLayer is not done in vtkImageView3D neither. That's why i was asking if you saw a data or a process leading to GetImage2DDisplayForLayer being null, and if it can be solved :)